src/approx_const.rs

```diff
@@ -31,6 +32,13 @@ impl LintPass for ApproxConstant {
 
     fn check_expr(&mut self, cx: &Context, e: &Expr) {
         if let &ExprLit(ref lit) = &e.node {
+            if let Some(res) = cx.tcx.def_map.borrow().get(&e.id) {
+                let krate = res.def_id().krate;
+                let cdata = &cx.sess().cstore.get_crate_data(krate);
```

Comment:
This is a heavy call, IIRC (most of the metadata calls are). Should not be first.